OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/CityControllerTests.cs

```diff
@@ -0,0 +1,266 @@
+
+            // Assert
+            Assert.IsNotNull(result);
+            Assert.AreEqual(BadRequestStatusCode, result.StatusCode);
+        }
+
+        [Test]
+        [TestCase("Test")]
+        public async Task GetCityByName_WhenNameNotExist_ReturnsNoContentResult(string name)
+        {
+            // Arrange
+            service.Setup(x => x.GetByCityName(name)).ReturnsAsync(cities.Where(x => x.Name.Contains(name)));
+
+            // Act
+            var result = await controller.GetByName(name).ConfigureAwait(false) as NoContentResult;
+
+            // Assert
+            Assert.IsNotNull(result);
+            Assert.AreEqual(NoContentStatusCode, result.StatusCode);
+        }
+
+        [Test]
+        public async Task CreateCity_WhenModelIsValid_ReturnsCreatedAtActionResult()
+        {
+            // Arrange
+            service.Setup(x => x.Create(city)).ReturnsAsync(city);
+
+            // Act
+            var result = await controller.Create(city).ConfigureAwait(false) as CreatedAtActionResult;
+
+            // Assert
+            Assert.IsNotNull(result);
+            Assert.AreEqual(CreateStatusCode, result.StatusCode);
+        }
+
+        [Test]
+        public async Task UpdateCity_WhenModelIsValid_ReturnsOkObjectResult()
+        {
+            // Arrange
+            service.Setup(x => x.Update(city)).ReturnsAsync(city);
+
+            // Act
+            var result = await controller.Update(city).ConfigureAwait(false) as OkObjectResult;
+
+            // Assert
+            Assert.IsNotNull(result);
+            Assert.AreEqual(OkStatusCode, result.StatusCode);
+        }
+
+        [Test]
+        [TestCase(1)]
+        public async Task DeleteCity_WhenIdIsValid_ReturnsNoContentResult(long id)
+        {
+            // Arrange
+            service.Setup(x => x.Delete(id));
+
+            // Act
+            var result = await controller.Delete(id) as NoContentResult;
+
+            // Assert
+            Assert.IsNotNull(result);
+            Assert.AreEqual(NoContentStatusCode, result.StatusCode);
+        }
+
+        [Test]
+        [TestCase(-100)]
+        public void DeleteCity_WhenIdIsInvalid_ReturnsArgumentOutOfRangeException(long id)
+        {
+            // Arrange
+            service.Setup(x => x.Delete(id));
+
+            // Act and Assert
+            Assert.ThrowsAsync<ArgumentOutOfRangeException>(
+                async () => await controller.Delete(id).ConfigureAwait(false));
+        }
+
+        [Test]
+        [TestCase(10)]
+        public async Task DeleteCity_WhenIdIsInvalid_ReturnsNull(long id)
+        {
+            // Arrange
+            service.Setup(x => x.Delete(id));
+
+            // Act
+            var result = await controller.Delete(id) as OkObjectResult;
+
+            // Assert
+            Assert.IsNull(result);
+        }
+
+        private CityDto FakeCity()
+        {
```

Comment:
can be made static)

Reply:
Dear @yfedo, 
Thank you for your advice.
This was added to the project.
Thank you.